### PR TITLE
192 / Activity feed

### PR DESCRIPTION
**Pull request by elliotwesoff**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# [Optional] Are there any post-deployment tasks we need to perform?

### CODE REVIEW

**Comment by elliotwesoff** [comment]
Updated screenshot

File: packages/db/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "@t3-oss/env-nextjs": "^0.7.1",
     "dotenv-cli": "^7.3.0",
     "eslint": "^8.53.0",
-    "tsx": "^3.9.0",
+    "tsx": "4.7.2",
```
Comment (mva919) [inline packages/db/package.json:40]: Why was this changed?

**Review by mva919 — APPROVED** [review]
Works on my end

**Comment by mva919** [comment]
Thank you for figuring out how to make connections in the mock data, this will come in clutch 💯

File: packages/db/package.json (hunk 1 above)
Comment (thienguen) [reply packages/db/package.json:40]: https://github.com/backstage/backstage/issues/21738

refer to this

File: apps/expo/src/app/activity/feed.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const { data: friends, isLoading: friendsIsLoading } = api.user.getFriends.useQuery({
+    userId: userData?.id ?? 0,
+  })
+  const { data: friendsActivities, isLoading: friendsActivitiesLoading } =
+    api.workout.getActivityFeedWorkouts.useQuery(
+      {
+        friendIds: friends?.map((friend) => friend.friendId) ?? [],
+        count: ACTIVITY_FEED_ITEM_LIMIT,
+      },
+      { enabled: !friendsIsLoading },
```
Comment (thienguen) [inline apps/expo/src/app/activity/feed.tsx:22]: trpc convention for the `enabled` query you gotta use !! operator for more edge case handling. You can make a commit to update, or nein, as it should work for this particular query ...

File: packages/api/src/router/user.router.ts [hunk 3]
```diff
@@ -66,6 +66,25 @@ export const userRouter = createTRPCRouter({
     })
   }),
 
+  /**
+    * get a user's friends
+    */
```
Comment (thienguen) [inline packages/api/src/router/user.router.ts:71]: this could be place in the friend router instead, it called the `friend` collection.

File: packages/api/src/router/user.router.ts
Comment (thienguen) [inline packages/api/src/router/user.router.ts:1]: refer to the previous comments

**Review by thienguen — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -57,6 +57,8 @@ yarn-error.log*
 # expo simulator build files
 apps/expo/android
 apps/expo/ios
+android
```
[the remark this answers is not in the document]
Comment (elliotwesoff) [reply .gitignore:60]: Those folders were generated at the root level of the repo. I'm not sure how they got there 🤷

File: apps/expo/src/app/activity/feed.tsx (hunk 2 above)
Comment (elliotwesoff) [reply apps/expo/src/app/activity/feed.tsx:22]: Well `!!` converts a truthy/falsey value to `true`/`false`. I want the opposite of that, so `!` should be sufficient.

File: packages/api/src/router/user.router.ts (hunk 3 above)
Comment (elliotwesoff) [reply packages/api/src/router/user.router.ts:71]: Will update tonight. Thanks!

File: apps/expo/src/app/activity/feed.tsx (hunk 2 above)
Comment (thienguen) [reply apps/expo/src/app/activity/feed.tsx:22]: if its logic then great, I overlooked it
